include/hc_am.hpp

```diff
@@ -174,6 +174,28 @@ void am_memtracker_update_peers(const hc::accelerator &acc, int peerCnt, hsa_age
  */
 am_status_t am_map_to_peers(void* ptr, size_t num_peer, const hc::accelerator* peers); 
 
+/*
+ * Locks a host pointer to a vector of agents
+ * 
+ * @p ac acclerator corresponding to current device
+ * @p hostPtr pointer to host memory which should be page-locked
+ * @p size size of hostPtr to be page-locked
+ * @p accVec vector to hsa agents to which the hostPtr should be visible
+ * @return AM_SUCCESS if lock is successfully.
+ * @return AM_ERROR_MISC if lock is unsuccessful.
+ */
+am_status_t am_memory_host_lock(hc::accelerator &ac, void *hostPtr, size_t size, std::vector<hsa_agent_s> &accVec);
```

Comment:
Right - for an HC interface we should use a vector of accelerators rather than agents.  Then in the lock implementation create the agent list from the accelerator (there is an hc::accelerator:: get_hsa_agent() that might help here).
Also if does the agent where the pointer is allocated also need to be on the visibility list?  I think the answer is YES – we should make this more clear in the description.
-ben

From: Jack Chung [mailto:notifications@github.com]
Sent: Thursday, April 14, 2016 12:08 PM
To: RadeonOpenCompute/hcc <hcc@noreply.github.com>
Subject: Re: [RadeonOpenCompute/hcc] added feature for locking host memory (#24)


In include/hc_am.hpp<https://github.com/RadeonOpenCompute/hcc/pull/24#discussion_r59754156>:

> @@ -174,6 +174,28 @@ void am_memtracker_update_peers(const hc::accelerator &acc, int peerCnt, hsa_age

>   */

>  am_status_t am_map_to_peers(void* ptr, size_t num_peer, const hc::accelerator* peers);

>

> +/*

> + * Locks a host pointer to a vector of agents

> + *

> + * @p ac acclerator corresponding to current device

> + * @p hostPtr pointer to host memory which should be page-locked

> + * @p size size of hostPtr to be page-locked

> + * @p accVec vector to hsa agents to which the hostPtr should be visible

> + * @return AM_SUCCESS if lock is successfully.

> + * @return AM_ERROR_MISC if lock is unsuccessful.

> + */

> +am_status_t am_memory_host_lock(hc::accelerator &ac, void *hostPtr, size_t size, std::vector<hsa_agent_s> &accVec);

could you consider remove the dependency of hsa_agent_s here? for example, change them to std::vector ? normally we want to avoid having direct HSA dependency in HCC public headers, so it's possible to build x86 simulation path or OpenCL path.

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly or view it on GitHub<https://github.com/RadeonOpenCompute/hcc/pull/24/files/b118eec1e4e1d03df2beb97d06eb3f40874a271c#r59754156>